core/systems/disks/steam_removable_media_manager.gd

```diff
@@ -0,0 +1,241 @@
+	var sd_card: BlockDevice
+	for block in device_tree:
+		if not block.dbus_path.contains("mmcblk0"):
+			continue
+		sd_card = block
+	if not sd_card:
+		logger.warn("Unable to find SD Card!")
+		return
+	if _block_device_has_protected_mount(sd_card):
+		logger.warn("Attempted to format SD Card with protected mount!")
+		return
+
+	logger.debug("Formatting SD Card: mmcblk0")
+	var result = await _execute_in_thread(STEAMOS_FORMAT_SDCARD)
+	logger.debug("Got results: " + str(result[0]), str(result[1]))
+
+	var success: bool = result[1] == 0
+	sd_card.format_complete.emit(success)
+
+
+## Calls the SteamRemovableMedia or SteamOS trim-devices script to perform a
+## trim operation on mmcblk0.
+func trim_sd_card() -> void:
+	if not trim_capable:
+		logger.error("System is not trim capable")
+		return
+
+	# Make sure the sd card isn't being used as a protected drive.
+	var sd_card: BlockDevice
+	for block in device_tree:
+		if not block.dbus_path.contains("mmcblk0"):
+			continue
+		sd_card = block
+	if not sd_card:
+		logger.warn("Unable to find SD Card!")
+		return
+	if _block_device_has_protected_mount(sd_card):
+		logger.warn("Attempted to format SD Card with protected mount!")
+		return
+
+	logger.debug("Performing TRIM operation on SD Card: mmcblk0")
+	var result = await _execute_in_thread(STEAMOS_TRIM_DEVICES)
+	logger.debug("Got results: " + str(result[0]), str(result[1]))
+
+	var success: bool = result[1] == 0
+	sd_card.trim_complete.emit(success)
+
+
+## Finds all partitions of the given block device and returns true if any of
+## them have mounts in the protected_mounts list.
+func _block_device_has_protected_mount(device: BlockDevice) -> bool:
+	logger.debug("Checking block device", device.dbus_path, "for protected mounts.")
+	# Get all the partition dbus paths of this block device
+	for partition in device.partitions:
+		if _partition_has_protected_mount(partition):
+			return true
+	return false
+
+
+## Loops through all mount points of the given partition and returns true if any of
+## them have mounts in the protected_mounts list.
+func _partition_has_protected_mount(device: PartitionDevice) -> bool:
+	logger.debug("Checking partition", device.dbus_path, "for protected mounts.")
+	for mount_point in device.fs_mount_points:
+		logger.debug("Checking mount point", mount_point, "for protected mounts.")
+		if mount_point in protected_mounts:
+			logger.debug("Found a protected mount point on drive: " + device.dbus_path)
+			return true
+	return false
+
+
+
+func _execute_in_thread(path: String, args: Array = []) -> Array:
+	var thread: SharedThread = SharedThread.new()
+	thread.watchdog_enabled = false
+	thread.start()
+	var result = await thread.exec(_exec_call.bind(path, args))
+	thread.stop()
+	return result
+
+
+## Executes the given command and arguments
+func _exec_call(path: String, args: Array = []) -> Array:
+	var output = []
+	var exit_code := OS.execute(path, args, output)
+	return [output, exit_code]
```

Comment:
Instead of duplicating this, you can use the `Command` class to execute a command in a thread.